tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java

```diff
@@ -0,0 +1,266 @@
+ *  under the License.
+ */
+
+package org.ballerinalang.test.documentation;
+
+import org.ballerinalang.docgen.docs.BallerinaDocGenerator;
+import org.ballerinalang.docgen.generator.model.Constant;
+import org.ballerinalang.docgen.generator.model.Construct;
+import org.ballerinalang.docgen.generator.model.Error;
+import org.ballerinalang.docgen.generator.model.FiniteType;
+import org.ballerinalang.docgen.generator.model.Function;
+import org.ballerinalang.docgen.generator.model.Module;
+import org.ballerinalang.docgen.generator.model.Object;
+import org.ballerinalang.docgen.generator.model.Project;
+import org.ballerinalang.docgen.generator.model.Record;
+import org.ballerinalang.docgen.generator.model.UnionType;
+import org.ballerinalang.docgen.model.ModuleDoc;
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test cases to check @deprecated annotation is showing in the docs.
+ */
+public class DeprecatedAnnotationTest {
+
+    private Module testModule;
+
+    @BeforeClass
+    public void setup() throws IOException {
+        String sourceRoot =
+                "test-src" + File.separator + "documentation" + File.separator + "deprecated_annotation_project";
+        CompileResult result = BCompileUtil.compile(sourceRoot, "test_module");
+
+        List<BLangPackage> modules = new LinkedList<>();
+        modules.add((BLangPackage) result.getAST());
+        Map<String, ModuleDoc> docsMap = BallerinaDocGenerator.generateModuleDocs(
+                Paths.get("src/test/resources", sourceRoot).toAbsolutePath().toString(), modules);
+        List<ModuleDoc> moduleDocList = new ArrayList<>(docsMap.values());
+        moduleDocList.sort(Comparator.comparing(pkg -> pkg.bLangPackage.packageID.toString()));
+        Map<String, List<Path>> resources = new HashMap<>();
+
+        Project project = BallerinaDocGenerator.getDocsGenModel(moduleDocList, resources);
+        testModule = project.modules.get(0);
+    }
+
+    @Test(description = "Test @deprecated annotation for module-level union type definitions")
+    public void testDeprecatedUnionTypeDef() {
+        List<UnionType> unionTypes = testModule.unionTypes;
+        UnionType depUnionType = null;
+        UnionType nonDepUnionType = null;
+
+        for (UnionType type : unionTypes) {
+            String typeName = type.name;
+            if ("CountryCode".equals(typeName)) {
+                depUnionType = type;
+            } else if ("NewCountryCode".equals(typeName)) {
```

Comment:
```suggestion
            } else if (typeName.equals("NewCountryCode")) {
```

Reply:
In the repo, we have used it in this way. That's why I used the same convention.